infrastructure/main.tf

```diff
@@ -0,0 +1,145 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 5.0"
+    }
+  }
+
+  required_version = ">= 1.0.0"
+}
+
+provider "aws" {
+  shared_credentials_files = ["~/.aws/credentials"]
+  profile                  = "igork_aws"
+  default_tags {
+    tags = {
+      Owner = "Igor Kostin"
+    }
+  }
+}
+
+locals {
+  subnet_prefix_name = {
+    public  = "public_SUBENT_${var.owner_code_name}_${var.vpc_version}"
+    private = "private_SUBENT_${var.owner_code_name}_${var.vpc_version}"
+    db      = "db_SUBENT_${var.owner_code_name}_${var.vpc_version}"
+  }
+
+  subnet_cider_bits = {
+    public  = 8
+    private = 2
+    db      = 8
+  }
+
+  vpc_cider_block = var.vpc_cidr_block
+}
+
+variable "azs" {
+  type        = list(string)
+  description = "Availability Zones"
+  default     = ["eu-central-1a", "eu-central-1b", "eu-central-1c"]
+}
+
+module "public_subnet_cider_blocks" {
+  source = "hashicorp/subnets/cidr"
+
+  base_cidr_block = var.vpc_cidr_block
+
+  networks = [
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.private}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.private)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.db}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.db)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[0]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[1]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    },
+    {
+      name     = "${local.subnet_prefix_name.public}_${var.azs[2]}"
+      new_bits = tonumber(local.subnet_cider_bits.public)
+    }
+  ]
+}
+
+locals {
```

Comment:
Not needed, too complicated, in the VPC module I'll give you example how to solve this.